lib/executor/ramping_arrival_rate_test.go

```diff
@@ -243,6 +243,60 @@ func TestRampingArrivalRateRunCorrectRateWithSlowRate(t *testing.T) {
 	assert.Equal(t, int64(2), es.GetInitializedVUsCount())
 }
 
+func BenchmarkRampingArrivalRateRun(b *testing.B) {
+	tests := []struct {
+		prealloc null.Int
+	}{
+		{prealloc: null.IntFrom(100)},
+		{prealloc: null.IntFrom(1000)},
+	}
+
+	for _, tc := range tests {
+		b.Run(fmt.Sprintf("PreAllocatedVUs%d", tc.prealloc.ValueOrZero()), func(b *testing.B) {
+			engineOut := make(chan stats.SampleContainer, 100)
```

Comment:
I suggest you increase the size of this buffer, as well as add a goroutine that reads from it and discards any metrics (we don't care about them in the benchmark). Otherwise, if your test emits too many `dropped_iteration` metrics, this will fill up and the test will deadlock like this:
```
$ go test -trimpath -benchmem -run='^$' -bench "BenchmarkRampingArrivalRateRun" github.com/loadimpact/k6/lib/executor

fatal error: all goroutines are asleep - deadlock!

goroutine 1 [chan receive]:
testing.(*B).run1(0xc000146900, 0xc000146900)
	testing/benchmark.go:234 +0xa5
testing.(*B).Run(0xc0001466c0, 0xae7a72, 0x1e, 0xb06678, 0x508000)
	testing/benchmark.go:667 +0x373
testing.runBenchmarks.func1(0xc0001466c0)
	testing/benchmark.go:541 +0x78
testing.(*B).runN(0xc0001466c0, 0x1)
	testing/benchmark.go:192 +0xeb
testing.runBenchmarks(0xaec487, 0x25, 0xc00000c558, 0x1043060, 0x5, 0x5, 0x104d220)
	testing/benchmark.go:550 +0x3d1
testing.(*M).Run(0xc000189880, 0x0)
	testing/testing.go:1423 +0x51d
main.main()
	_testmain.go:131 +0x138

goroutine 33 [chan receive]:
testing.(*B).run1(0xc000146b40, 0xc000146b40)
	testing/benchmark.go:234 +0xa5
testing.(*B).Run(0xc000146900, 0xc0000281b0, 0x12, 0xc000058000, 0x0)
	testing/benchmark.go:667 +0x373
github.com/loadimpact/k6/lib/executor.BenchmarkRampingArrivalRateRun(0xc000146900)
	github.com/loadimpact/k6/lib/executor/ramping_arrival_rate_test.go:255 +0x188
testing.(*B).runN(0xc000146900, 0x1)
	testing/benchmark.go:192 +0xeb
testing.(*B).run1.func1(0xc000146900)
	testing/benchmark.go:232 +0x57
created by testing.(*B).run1
	testing/benchmark.go:225 +0x7f

goroutine 9 [chan send]:
github.com/loadimpact/k6/stats.PushIfNotDone(...)
	github.com/loadimpact/k6/stats/stats.go:434
github.com/loadimpact/k6/lib/executor.RampingArrivalRate.Run(0xc0000763c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/loadimpact/k6/lib/executor/ramping_arrival_rate.go:473 +0x1485
github.com/loadimpact/k6/lib/executor.BenchmarkRampingArrivalRateRun.func1(0xc000146b40)
	github.com/loadimpact/k6/lib/executor/ramping_arrival_rate_test.go:290 +0x62f
testing.(*B).runN(0xc000146b40, 0x1)
	testing/benchmark.go:192 +0xeb
testing.(*B).run1.func1(0xc000146b40)
	testing/benchmark.go:232 +0x57
created by testing.(*B).run1
	testing/benchmark.go:225 +0x7f

goroutine 10 [chan receive]:
github.com/loadimpact/k6/lib/executor.RampingArrivalRate.Run.func4(0xc0000322a0, 0xc0000223e8, 0xc00018a060, 0xc000032420, 0xc0000780d0, 0xba23a0, 0xc00018a000, 0xc00006c0f0)
	github.com/loadimpact/k6/lib/executor/ramping_arrival_rate.go:380 +0xc5
created by github.com/loadimpact/k6/lib/executor.RampingArrivalRate.Run
	github.com/loadimpact/k6/lib/executor/ramping_arrival_rate.go:371 +0xb29

goroutine 244 [chan send]:
github.com/loadimpact/k6/lib/executor.RampingArrivalRateConfig.cal(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/loadimpact/k6/lib/executor/ramping_arrival_rate.go:274 +0x411
created by github.com/loadimpact/k6/lib/executor.RampingArrivalRate.Run
	github.com/loadimpact/k6/lib/executor/ramping_arrival_rate.go:444 +0x10f9
```